Add document for slim demo.

wanghaoshuang opened the pull request.

wanghaoshuang committed:
Add README.md

wanghaoshuang committed:
Update url of images.

wanghaoshuang committed:
Add images.

wanghaoshuang committed:
Add usage.md

wanghaoshuang committed:
Add demo.md

wanghaoshuang committed:
Fix demo.md

wanghaoshuang committed:
Add model_zoo.md

wanghaoshuang committed:
Fix pretrain models download in model zoo.

qingqing01 commented on fluid/PaddleSlim/README.md:
```diff
@@ -0,0 +1,170 @@
+<strong>图 1</strong>
+</p>
+
+如**图 1**所示，最上层的紫色模块为用户接口，在python脚本中调用模型压缩功能时，只需要构造一个Compressor对象即可，在[使用文档](docs/usage.md)中会有详细说明。
```
python -> Python

wanghaoshuang replied:
Thx. Fixed.

qingqing01 commented on fluid/PaddleSlim/README.md:
```diff
@@ -0,0 +1,170 @@
+### 剪切
+
+- 支持敏感度和uniform两种方式
+- 支持vgg、resnet、mobilenet等各种类型的网络
```
vgg -> VGG
resnet -> ResNet
mobilenet -> MobileNet

wanghaoshuang replied:
Thx. Fixed.

qingqing01 commented on fluid/PaddleSlim/README.md:
```diff
@@ -0,0 +1,170 @@
+- 支持动态和静态两种量化训练方式
+- 支持以float类型模拟int8保存模型
+- 支持以int8类型保存模型
+- 支持以兼容paddle mobile的格式保存模型
```
可以在首页最后加:   ### 模型导出格式

里面介绍下支持的导出格式

wanghaoshuang replied:
Thx. Done.

qingqing01 commented on fluid/PaddleSlim/README.md:
```diff
@@ -0,0 +1,170 @@
+数据：ImageNet 1000类
+模型：MobileNetV1
+
+| 剪切flops |剪切精度（top5/top1） |剪切+量化（dynamic）
```
flops -> FLOPs

wanghaoshuang replied:
Thx. Fixed.

qingqing01 commented on fluid/PaddleSlim/docs/demo.md:
```diff
@@ -0,0 +1,280 @@
+
+脚本run.sh会自动从[models/fluid/PaddleCV/image_classification](https://github.com/PaddlePaddle/models/tree/develop/fluid/PaddleCV/image_classification#supported-models-and-performances)下载ResNet50和MobileNetV1的预训练模型，并放入PaddleSlim/pretrain路径下。
+
+>注意：下载的预训练模型是‘specify parameter names’的，也就是模型中变量都是指定的名称，而不是随机生成的名称。
```
去掉这段，分类里已去掉

wanghaoshuang replied:
Thx. Fixed.

qingqing01 commented on fluid/PaddleSlim/README.md:
```diff
@@ -0,0 +1,170 @@
+### 剪切后量化
+
+数据：ImageNet 1000类
+模型：MobileNetV1
```
组合实验是否需要同时放在一个表格里？

两个表格对比，既有参量，又有精度。

wanghaoshuang replied:
Thx. Done.

qingqing01 commented on fluid/PaddleSlim/README.md:
```diff
@@ -0,0 +1,170 @@
+- 以配置文件方式集中管理可配参数，方便实验管理
+- 在普通模型训练脚本上，添加极少代码即可完成模型压缩
+
+详见：[使用示例](docs/demo.md)
```
易用性，首页里是否要展示一个最简单的示例，几行即可运行起来?

wanghaoshuang replied:
如果在首页加例子的话，几行代码还搞不定，所以感觉还是不加为好。

wanghaoshuang committed:
Merge branch 'develop' of https://github.com/PaddlePaddle/models into add_doc

wanghaoshuang committed:
Refine documents.
1. Rename python to Python
2. Fix README.md: vgg->VGG, resnet->ResNet, mobilenet->MobileNet
3. Rename flops to FLOPS

wanghaoshuang committed:
Move docs of PaddleSlim from models.fluid.PaddleSlim to models.PaddleSlim

wanghaoshuang committed:
1. Introduce model format in README.md
2. Merge results of combined strategies into one table.

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,173 @@
+
+- **普通Paddle格式：** 普通Paddle模型格式，可通过Paddle框架加载使用。
+- **模拟int8的float数据格式模型：** 仅在量化训练策略时使用，用普通的Paddle模型格式（即float数据类型）模拟保存int8量化后的int8模型。可通过Paddle框架直接加载使用，一般用于验证int8量化后模型的精度。
+- **int8数据格式模型：** 仅在量化训练策略时使用，直接用int8数据类型保存模型权重。
```
去掉 "模拟int8的float数据格式模型" 和 “int8数据格式模型”，保留2个就行了: Paddle Fluid模型格式， Paddle Mobile模型格式

wanghaoshuang replied:
Thx. Done.

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,173 @@
+
+- 支持动态和静态两种量化训练方式
+- 支持以float类型模拟int8保存模型
+- 支持以int8类型保存模型
```
建议去掉line 92和93，换成：

支持对权重全局量化和Channel-Wise量化 ？  @wzzju 看下这块描述

wanghaoshuang replied:
Thx. Done.

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,173 @@
+- 支持动态和静态两种量化训练方式
+- 支持以float类型模拟int8保存模型
+- 支持以int8类型保存模型
+- 支持以兼容paddle mobile的格式保存模型
```
paddle mobile -> Paddle Mobile

wanghaoshuang replied:
Thx. Done.

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,173 @@
+
+### 量化训练
+
+- 支持动态和静态两种量化训练方式
```
这里解释下 动态和静态 区别？

wanghaoshuang replied:
Thx. Done.

wanghaoshuang committed:
Refine README.md

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,173 @@
+- [架构介绍](#架构介绍)
+- [功能列表](#功能列表)
+- [实验结果](#简要实验结果)
+- [模型导出格式](#模型导出格式)
```
之前讨论，Model Zoo在首页文档里没有体现到

wanghaoshuang replied:
将‘实验结果’修改为了‘实验结果与ModelZoo’, 并在首页实验结果章节给出了到ModelZoo的链接。

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,173 @@
+- [实验结果](#简要实验结果)
+- [模型导出格式](#模型导出格式)
+
+## 特色
```
特色 -> 主要特点

wanghaoshuang replied:
Fixed.

wanghaoshuang committed:
Refine README
1. Fix some terms in README.md
2. Fix description of model format in README.md
3. Add Channel-Wise into feature list
4. Rename 'paddle mobile' to 'Paddle Mobile'
5. Add comments for dynamic quantization and static quantization
6. Add entry url for Model Zoo in README.md

wzzju commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,176 @@
+| Model | FP32| int8(A:abs_max, W:abs_max) | int8, (A:moving_average_abs_max, W:abs_max) |int8, (A:abs_max, W:channel_wise_abs_max) |
+|:---|:---:|:---:|:---:|:---:|
+|MobileNetV1|70.916%|71.008%|70.84%|71.00%|
+|ResNet50|76.352%|76.612%|76.456%|76.73%|
```
Please update the quantization result, refer to http://agroup.baidu.com/paddlecv/md/article/1642698.

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,176 @@
+## 架构介绍
+
+这里简要介绍模型压缩工具实现的整体原理，便于理解使用流程。
+**图 1**为模型压缩工具的架构图，从上到下为API依赖关系。蒸馏模块、量化模块和剪切模块都间接依赖底层的paddle框架。考虑到这种依赖关系和版本的管理，我们将模型压缩工具作为了paddle框架的一部分，所以已经安装普通版本paddle的用户需要重新下载安装支持模型压缩功能的paddle，才能使用压缩功能。
```
paddle -> PaddlePaddle

考虑到这种依赖关系和版本的管理，我们将模型压缩工具作为了paddle框架的一部分  ->  目前，模型压缩工具作为了PaddlePaddle框架的一部分

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,176 @@
+---
+# PaddleSlim模型压缩工具库
+
+PaddleSlim是PaddlePaddle框架的一个子模块。PaddleSlim首次在PaddlePaddle 1.4版本中发布。在PaddleSlim中，实现了目前主流的网络剪枝、参数量化、模型蒸馏三种压缩策略，主要用于压缩图像领域模型。在后续版本中，会添加更多的压缩策略，以及完善对NLP领域模型的支持。
```
参数量化 ->  量化
模型蒸馏 ->  蒸馏

参数，会让人误以为只针对weight

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,176 @@
+### 量化训练
+
+- 支持动态和静态两种量化训练方式
+  - 动态策略: 在每次迭代过程中动态统计并使用量化参数。
```
推理过程动态统计激活的量化参数

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,176 @@
+
+- 支持动态和静态两种量化训练方式
+  - 动态策略: 在每次迭代过程中动态统计并使用量化参数。
+  - 静态策略: 对不同的输入，采用相同的从训练数据中统计得到的量化参数。
```
推理过程，对不同的输入，...

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,176 @@
+
+评估实验所使用数据集为ImageNet1000类数据，且以top-1准确率为衡量指标：
+
+| Model | FP32| int8(A:abs_max, W:abs_max) | int8, (A:moving_average_abs_max, W:abs_max) |int8, (A:abs_max, W:channel_wise_abs_max) |
```
A -> X ?

qingqing01 commented on PaddleSlim/README.md:
```diff
@@ -0,0 +1,176 @@
+压缩框架支持导出以下格式的模型：
+
+- **普通Paddle格式：** 普通Paddle模型格式，可通过Paddle框架加载使用。
+- **Paddle Mobile数据格式模型：** 仅在量化训练策略时使用，兼容[Paddle Mobile](https://github.com/PaddlePaddle/paddle-mobile)的模型格式。
```
普通Paddle格式  -> Paddle Fluid模型格式
Paddle Mobile数据格式模型 -> Paddle Mobile模型格式

wanghaoshuang committed:
Refine README.
